### ISSUE TITLE
Return error on invalid job ID

### ISSUE THREAD

**Issue by catthehacker** [rhysd/actionlint#80]
https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_id

> Create an identifier for your job by giving it a unique name. The key `job_id` is a string and its value is a map of the job's configuration data. You must replace `<job_id>` with a string that is unique to the `jobs` object. The `<job_id>` must start with a letter or `_` and contain only alphanumeric characters, `-`, or `_`.

Examples:

<details>
<summary>Invalid job name 1</summary>

```yaml
name: invalid-job-name-1
on: push

jobs:
  invalid-JOB-Name-v1.2.3-docker_hub:
    runs-on: ubuntu-latest
    steps:
      - run: echo hi
  valid-JOB-Name-v123-docker_hub:
    runs-on: ubuntu-latest
    steps:
      - run: echo hi
```
</details>

<details>
<summary>Invalid job name 2</summary>

```yaml
name: invalid-job-name-2
on: push

jobs:
  1234invalid-JOB-Name-v123-docker_hub:
    runs-on: ubuntu-latest
    steps:
      - run: echo hi
```
</details>

<details>
<summary>Valid job name 1</summary>

```yaml
name: valid-job-name-1
on: push

jobs:
  valid-JOB-Name-v123-docker_hub:
    runs-on: ubuntu-latest
    steps:
      - run: echo hi
```

</details>

<details>
<summary>Valid job name 2</summary>

```yaml
name: valid-job-name-2
on: push

jobs:
  ___valid-JOB-Name-v123-docker_hub:
    runs-on: ubuntu-latest
    steps:
      - run: echo hi
```

</details>
